testutils/resourcebuilder/machine/v1beta1/aws_provider_spec.go

```diff
@@ -127,26 +154,76 @@ func (m AWSProviderSpecBuilder) BuildRawExtension() *runtime.RawExtension {
 	}
 }
 
+// WithAMI sets the AMI for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithAMI(ami *machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
+	m.ami = ami
+	return m
+}
+
 // WithAvailabilityZone sets the availabilityZone for the AWS machine config builder.
-func (m AWSProviderSpecBuilder) WithAvailabilityZone(az string) AWSProviderSpecBuilder {
+func (m AWSProviderSpecBuilder) WithAvailabilityZone(az *string) AWSProviderSpecBuilder {
 	m.availabilityZone = az
```

Comment:
Why not take the address of AZ rather than changing the signature?

Reply:
Yes good point! Updated. LMK if the change sounds reasonable and I can extend more fields like so.